src/spdx2/agent/spdx2.php

```diff
@@ -717,8 +717,9 @@ protected function generateFileNodesByFiles($filesWithLicenses, $treeTableName,
       {
         $licenses['scanner'] = array();
       }
-      if(!$this->getIgnoreFilesWOinfo($uploadId) ||
-         ($this->getIgnoreFilesWOinfo($uploadId) && (!empty($licenses['concluded']) || (!empty($licenses['scanner']) && !empty($licenses['scanner'][0])) || !empty($licenses['copyrights']))))
+      $stateWoInfos = $this->getIgnoreFilesWOinfo($uploadId);
+      if(!$stateWoInfos ||
+         ($stateWoInfos && (!empty($licenses['concluded']) || (!empty($licenses['scanner']) && !empty($licenses['scanner'][0])) || !empty($licenses['copyrights']))))
```

Comment:
Actually, I meant that the logic can be simplified :
`(!$stateWoInfos || ($stateWoInfos && (!empty($licenses['concluded']) || ...))`

Can be simplified into:
`(!$stateWoInfos || (!empty($licenses['concluded']) || ...)`

Reply:
Not the same: empty tests must be done only when stateWoInfos is true.

Reply:
OK then, sorry.